forever.go

```diff
@@ -24,14 +25,25 @@ func (h *ForeverHandle) WaitUntilShutdown(timeoutCtx context.Context) {
 	}
 }
 
+func (h *ForeverHandle) waitUntilSupervised(ctx context.Context) {
+	select {
+	case <-h.supervisedChan:
+	case <-ctx.Done():
+		h.MarkSupervised() // context closed before supervision. not an error: mark as supervised to avoid logging as error
```

Comment:
If we mandate `waitUntilSupervised` as a prerequisite of launching `forever`, doesn't this make the check for `!h.supervised` in `terminated` redundant? we can avoid printing the error in all cases, because now this error becomes meaningless. the goroutine will never run anything if it's unsupervised, so we don't need to `MarkSupervised()` here as well